### PR TITLE
Add configuration warnings to SD

### PR DESCRIPTION
**Pull request by arogge**
This PR adds configuration warnings to the SD when
* `Maximum Block Size` is configured on a device that is not a tape
* The SD is invoked with `-p` to ignore i/o errors

### CODE REVIEW

File: core/src/stored/stored.cc [hunk 1]
```diff
@@ -253,6 +253,10 @@ int main(int argc, char* argv[])
   my_config = InitSdConfig(configfile, M_ERROR_TERM);
   ParseSdConfig(configfile, M_ERROR_TERM);
 
+  if (forge_on) {
+    my_config->AddWarning("Running with '-p' is for testing purposes only");
```
Comment (pstorz) [inline core/src/stored/stored.cc:257]: How about "for testing and emergency recovery purposes only"

File: core/src/stored/stored_conf.cc [hunk 2]
```diff
@@ -736,6 +736,11 @@ static bool SaveResource(int type, ResourceItem* items, int pass)
   int i;
   int error = 0;
 
+  BareosResource* allocated_resource = *resources[rindex].allocated_resource_;
+  if (pass == 2 && !allocated_resource->Validate()) {
+    return false;
```
Comment (pstorz) [inline core/src/stored/stored_conf.cc:741]: Does this return value mean that the storage daemon does not start?

**Review by pstorz — CHANGES_REQUESTED** [review]

File: core/src/stored/stored_conf.cc (hunk 2 above)
Comment (arogge) [reply core/src/stored/stored_conf.cc:741]: Yes. If Validate() returns false, the the SD will not start. This is analogous to the behaviour of the Director.
The idea is that we can run in-depth checks of the configuration that way (which was why that function added to the director in the first place).

File: core/src/stored/stored.cc (hunk 1 above)
Comment (arogge) [reply core/src/stored/stored.cc]: agreed.

**Pull request merged**